src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,167 @@
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+	}
+}
+
+// A C# Azure Functions project must be present in order to create a new Azure Function for the table
+async function isAzureFunctionProjectOpen(): Promise<boolean> {
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return false;
+	} else {
+		const hostFile = await getHostFile();
+		const projectFiles = await getProjectFile();
+		if (projectFiles !== undefined && hostFile !== undefined) {
+			// select project to add azure function to
+			selectedProjectFile = (await vscode.window.showQuickPick(projectFiles, {
+				canPickMany: false,
+				title: constants.selectProject,
+				ignoreFocusOut: true
+			}));
+			return selectedProjectFile !== undefined;
+		}
+		return;
+	}
+}
+
+// gets the azure functions project file path
+async function getProjectFile(): Promise<string[] | undefined> {
+	const projUris = await vscode.workspace.findFiles('**/*.csproj');
+	let projFiles: string[] = [];
+	for (let projUri of projUris) {
+		projFiles.push(projUri.fsPath);
+	}
+	return projFiles.length > 0 ? projFiles : undefined;
+}
+
+// gets the host file config file path
+async function getHostFile(): Promise<string | undefined> {
+	const hostFiles = await vscode.workspace.findFiles('**/host.json');
+	return hostFiles.length > 0 ? hostFiles[0].fsPath : undefined;
+}
+
+// gets the local.settings.json file path
+async function getSettingsFile(): Promise<string | undefined> {
+	const settingsFiles = await vscode.workspace.findFiles('**/local.settings.json');
+	return settingsFiles.length > 0 ? settingsFiles[0].fsPath : undefined;
+}
+
+// retrieves the new function file once the file is created
+function getNewFunctionFile(): Promise<string> {
+	return new Promise((resolve) => {
+		const watcher = vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(
+			vscode.workspace.workspaceFolders[0], '**/*.cs'), false, true, true);
+		watcher.onDidCreate((e) => {
+			resolve(e.fsPath);
+			watcher.dispose();
+		});
+	});
+}
+
+// adds the required nuget package to the project
+async function addNugetReferenceToProjectFile(): Promise<void> {
+	// Make sure the nuget source is added
+	const currentSources = await executeCommand('dotnet nuget list source');
+	if (currentSources.indexOf(sqlBindingNugetSource) === -1) {
+		await executeCommand(`dotnet nuget add source ${sqlBindingNugetSource}`);
+	}
+	await executeCommand(`dotnet add package ${constants.sqlExtensionPackageName} --prerelease`, path.dirname(selectedProjectFile));
```

Comment:
You should pass in the file path as a parameter, not use this global var.

Also do you need to do the dirname? dotnet add should take a direct reference to the project file just fine

Reply:
Yes it needs to be the dirname

Reply:
Oh, didn't realize you were passing this in as the cwd. You can just specify the project file as an arg

`dotnet add <PATH TO PROJECT> package Microsoft.Azure.WebJobs.Extensions.Sql --prerelease`